validation/validation.py

```diff
@@ -8,52 +8,56 @@
 
 PREDICTED_DAILY_NEW_CASES = "PredictedDailyNewCases"
 
-COLUMNS = ["CountryName",
+COLUMNS = {"CountryName",
            "RegionName",
            "Date",
-           PREDICTED_DAILY_NEW_CASES]
+           PREDICTED_DAILY_NEW_CASES}
 
 
 def validate_submission(start_date: str,
                         end_date: str,
-                        submission_url: str) -> List[str]:
+                        ip_file: str,
+                        submission_file: str) -> List[str]:
     """
     Checks a submission file is valid.
     Args:
         start_date: the submission start date as a string, format YYYY-MM-DDD
         end_date: the submission end date as a string, format YYYY-MM-DDD
-        submission_url: path to a file-like object
+        ip_file: path to a file-like object
+        submission_file: path to a file-like object
 
     Returns: a list of string messages if errors were detected, an empty list otherwise
 
     """
-    df = pd.read_csv(submission_url,
-                     parse_dates=['Date'],
-                     encoding="ISO-8859-1",
-                     error_bad_lines=True)
+    pred_df = pd.read_csv(submission_file,
+                          parse_dates=['Date'],
+                          encoding="ISO-8859-1",
+                          error_bad_lines=True)
+    ip_df = pd.read_csv(ip_file,
+                        parse_dates=['Date'],
+                        encoding="ISO-8859-1",
+                        error_bad_lines=True)
 
     all_errors = []
     # Check we go the expected columns
-    errors = _check_columns(COLUMNS, df)
-    if errors:
-        all_errors.extend(errors)
-    else:
-        # Columns are good, check the values in PredictedDailyNewCases
-        errors = _check_prediction_values(df)
-        if errors:
-            all_errors.extend(errors)
-        # Now check the prediction dates are correct
-        errors = _check_days(start_date, end_date, df)
-        if errors:
-            all_errors.extend(errors)
+    all_errors += _check_columns(COLUMNS, pred_df)
+    if not all_errors:
+        # Columns are good, check we got prediction for each requested country / region
+        all_errors += _check_geos(ip_df, pred_df)
+        # Check the values in PredictedDailyNewCases
+        all_errors += _check_prediction_values(pred_df)
+        # Check the prediction dates are correct
+        all_errors += _check_days(start_date, end_date, pred_df)
 
     return all_errors
 
 
-def _check_columns(expected_columns, df):
-    if not expected_columns == list(df.columns):
-        return [f"Not the expected list of columns. Expected columns are: {expected_columns}"]
-    return None
+def _check_columns(expected_columns, pred_df):
```

Comment:
Instead of checkin we have exactly the expected columns, we now check we have _at least_ the expected columns. It's fine to have more columns in the prediction file.